hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/InverseQuantiles.java

```diff
@@ -0,0 +1,44 @@
+package org.apache.hadoop.metrics2.util;
+
+import org.apache.hadoop.util.Preconditions;
+import java.util.ListIterator;
+
+public class InverseQuantiles extends SampleQuantiles{
+
+  public InverseQuantiles(Quantile[] quantiles) {
+    super(quantiles);
+  }
+  
+
+  /**
+   * Get the estimated value at the inverse of the specified quantile. 
+   * Eg: return the value at (1 - 0.99)*count position for quantile 0.99.
+   * When count is 100, quantile 0.99 is desired to return the value at the 1st position
+   *
+   * @param quantile Queried quantile, e.g. 0.50 or 0.99.
+   * @return Estimated value at the inverse position of that quantile. 
+   */
+  long query(double quantile) {
```

Comment:
im assuming this logic looks similar to the SampleQuantile, is there any small refactor we could do to DRY, then maybe a couple lines below looks like 
`int desired = getQuantile(quantile)` 
with a default and overridden implementation?

Reply:
agreed, that will be much cleaner. Let me make the change.

Reply:
- Minimal change here is to inverse list-order traversal. 
- The for loop is performing a Map operation. 

The DRY method would be to encapsulate the for loop to a protected function "getQuantilesFromSamples".

Reply:
Updated to reverse the traversal order.
Overriding `query()` method instead of creating a new method and moved out the common validation step outside.